src/ERC7540Vault.sol

```diff
@@ -49,6 +51,16 @@ contract ERC7540Vault is Auth, IERC7540 {
     /// @dev    Requests for Centrifuge pool are non-transferable and all have ID = 0
     uint256 constant REQUEST_ID = 0;
 
+    bytes32 private immutable nameHash;
+    bytes32 private immutable versionHash;
+    uint256 public immutable deploymentChainId;
```

Comment:
```
    bytes32 private immutable _NAME_HASH;
    bytes32 private immutable _VERSION_HASH;    
    bytes32 private immutable _DOMAIN_SEPARATOR;    
    uint256 public immutable DEPLOYMENT_CHAIN_ID;
    bytes32 public constant AUTHORIZE_OPERATOR_TYPEHASH = keccak256(
        "AuthorizeOperator(address owner,address operator,bool approved,uint256 validAfter,uint256 validBefore,bytes32 nonce)"
    );    
  ```
  
   Also I'm curious why `AUTHORIZED_OPERATOR_TYPEHASH` is initialized  here and the rest are initialized in the constructor?
   
   Having them initialized in the constructor is cheaper in overall cost.

Reply:
No good reason other than copying this over from the erc20 code where it's done similarly.

Reply:
> Having them initialized in the constructor is cheaper in overall cost.

Can you expand on this? Initializing in constructor is cheaper than initializing in the variable declaration?

Reply:
1. I think there should be consistency. One is initialized  in the variable declaration an some in the constructor. I think it should be in one place.
2. Naming should be also unified and I see some parts of the solidity style guide were followed and some not. We can agree of course on this :)
3. I was curious why  you would choose constructor initialization over in declaration so I deployed 2 version of the same contract ( on Remix ) .  
  - Version 1: All variables marked as constants and initialized during declaration
  - Version 2: All variables immutable and initialized in the constructor
 Version 2 deployment cost was less than the Version deployment cost.
 
 I don't know why it's that. it's just an observation.

Reply:
Let's put this in a ticket and decide how we want to handle this later